src/components/App/App.js

```diff
@@ -1,11 +1,44 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Navigate, Route, Routes } from 'react-router-dom';
+
+import { checkToken } from '../../redux/actions';
+import { currentUserSelector } from '../../redux/selectors';
+import ProtectedRoute from '../ProtectedRoute/ProtectedRoute';
+import Login from '../Login/Login';
+import NewOrder from '../NewOrder/NewOrder';
+import NotFound from '../NotFound/NotFound';
+import OrderDetails from '../OrderDetails/OrderDetails';
+import Preloader from '../Preloader/Preloader';
 
 import './App.css';
 
 function App() {
+
+  const dispatch = useDispatch();
+  const user = useSelector(currentUserSelector);
+
+  useEffect(() => {
+    dispatch(checkToken());
+  }, [dispatch]);
+
   return (
-    <div className="App">
-      App
+    <div className="app">
+      {
+        user.loading || user.loggedIn === null ?
+          <div className='app__preloader'>
+            <Preloader />
+          </div>
+          :
+          <Routes>
+            <Route element={<ProtectedRoute loggedIn={user.loggedIn} />} >
+              <Route path="/" element={<NewOrder />} />
+              <Route path="/order/:id" element={<OrderDetails />} />
+            </Route>
+            <Route path="/signin" element={user.loggedIn ? <Navigate replace to="/" /> : <Login />} />
+            <Route path="*" element={<NotFound />} />
+          </Routes>
+      }
```

Comment:
This is really minor change, but I would recommend to avoid ternary operators at the root levels of JSX markup just to make component look more clear:
```suggestion
    
useEffect(() => {
    dispatch(checkToken());
  }, [dispatch]);
  
  if (user.loading || user.loggedIn === null) {
    return <div className="app">
      <div className='app__preloader'>
        <Preloader />
      </div>
    </div>
  }

  return (
    <div className="app">
      <Routes>
        <Route element={<ProtectedRoute loggedIn={user.loggedIn} />} >
          <Route path="/" element={<NewOrder />} />
          <Route path="/order/:id" element={<OrderDetails />} />
        </Route>
        <Route path="/signin" element={user.loggedIn ? <Navigate replace to="/" /> : <Login />} />
        <Route path="*" element={<NotFound />} />
      </Routes>
    </div>
  );
```